src/events/start-blockchain-event-listeners.js

```diff
@@ -28,15 +28,10 @@ function startBlockchainEventListeners(eventCallbacks, startingBlockNumber, onSe
   blockStream.subscribeToOnLogAdded(subscriptions.onLogAdded);
   blockStream.subscribeToOnLogRemoved(subscriptions.onLogRemoved);
   async.forEachOf(eventCallbacks, function (callbacks, contractName, nextContract) {
```

Comment:
Is this loop still necessary?  If we are only ever adding one filter, it feels like we shouldn't be looping here?

Reply:
There's still 3. In the version you discussed in CH, it won't be necessary

Reply:
Not sure what other changes are queued up, but FWIW, `addFilter` currently calls `blockStream.addLogFilter` and sets up the callback (in augur.js) for each event.

Reply:
The change here is to to not do that by event, but by contract, to significantly reduce the number of getLogs requests per block. Micah is suggesting further changing to do it just once per block